packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -948,6 +993,37 @@ export function clearSuspenseBoundary(
         data === SUSPENSE_FALLBACK_START_DATA
       ) {
         depth++;
+      } else if (data.length === 1) {
```

Comment:
This may be a bit bold but rendering comments in React components it's super common and this is only run on dehydrated suspense boundaries so we're generally only susceptible to colliding with user of innerHTML. We can't eliminate every collision so while prefixing with `$` and this style of comment is not going to be written by a user so the most likely consequence is that there is a collision with an extension.

If we want to be significantly more defensive we can add a check that the nextNode is also a comment matching the Suspense end marker. But I don't think it's actually worth the code size to include

Reply:
I ended up making this more strict and only considering a comment a possible preamble contribution marker if it immediately precedes a suspense ending comment Node.